flink-catalog-examples/datagen/src/main/java/Datagen.java

```diff
@@ -0,0 +1,213 @@
+            System.exit(1);
+        }
+
+        final String filePath = cmd.getOptionValue("input") == null ? DEFAULT_FILE_PATH : cmd.getOptionValue("input");
+        File userBehaviorFile = new File(filePath);
+        final long speed = cmd.getOptionValue("speedup") == null ?
+                DEFAULT_SPEED : Long.parseLong(cmd.getOptionValue("speedup"));
+        final URI controllerUri = cmd.getOptionValue("controlleruri") == null ?
+                DEFAULT_CONTROLLER_URI : URI.create(cmd.getOptionValue("controlleruri"));
+        final URI schemaregistryUri = cmd.getOptionValue("schemaregistryuri") == null ?
+                DEFAULT_SCHEMAREGISTRY_URI : URI.create(cmd.getOptionValue("schemaregistryuri"));
+
+        ClientConfig clientConfig = ClientConfig.builder()
+                .controllerURI(controllerUri)
+                .build();
+        SchemaRegistryClientConfig schemaRegistryClientConfig = SchemaRegistryClientConfig.builder()
+                .schemaRegistryUri(schemaregistryUri)
+                .build();
+
+        try (StreamManager streamManager = StreamManager.create(controllerUri)){
+            streamManager.createScope(DEFAULT_SCOPE);
+            StreamConfiguration streamConfig = StreamConfiguration.builder()
+                    .build();
+            streamManager.createStream(DEFAULT_SCOPE, DEFAULT_STREAM, streamConfig);
+        } catch (Exception e) {
+            System.out.printf("Meet error when creating stream", e);
+            throw e;
+        }
+
+        // add schema registry group
+        GroupProperties groupProperties = new GroupProperties(SerializationFormat.Avro, Compatibility.allowAny(), false);
+        try (SchemaRegistryClient client = SchemaRegistryClientFactory.withNamespace(DEFAULT_SCOPE, schemaRegistryClientConfig)) {
+            if (!schemaGroupExist(client, DEFAULT_STREAM)) {
+                client.addGroup(DEFAULT_STREAM, groupProperties);
+            }
+        } catch (Exception e) {
+            System.out.printf("Meet error when adding schema registry group", e);
+            throw e;
+        }
+
+        // register schema to schema registry
+        SerializerConfig serializerConfig = SerializerConfig.builder()
+                .registryConfig(schemaRegistryClientConfig)
+                .namespace(DEFAULT_SCOPE)
+                .groupId(DEFAULT_STREAM)
+                .registerSchema(true)
+                .build();
+
+        Schema userBehavior = SchemaBuilder
+                .record("UserBehavior")
+                .fields()
+                .name("user_id").type(Schema.create(Schema.Type.INT)).noDefault()
+                .name("item_id").type(Schema.create(Schema.Type.INT)).noDefault()
+                .name("category_id").type(Schema.create(Schema.Type.INT)).noDefault()
+                .name("behavior").type(Schema.create(Schema.Type.STRING)).noDefault()
+                .name("ts").type(LogicalTypes.timestampMillis().
```

Comment:
It does not match with the string you write below for events.

Reply:
fixed